test/blackbox-tests/test-cases/variables-for-artifacts/run.t

```diff
@@ -243,5 +243,5 @@ field of a (rule).
   File "dune", line 3, characters 9-16:
   3 |  (deps %{cmo:x2})
                ^^^^^^^
-  Error: %{cmo:..} cannot be used in this position
+  Error: %{cmo:..} isn't allowed in this position
```

Comment:
Off-topic for the present PR, but with all the ongoing refactoring, is it/will it be possible to use these kinds of variables in the `deps` field? If memory serves they can already be used in `action` fields and they will be included automatically in the set of dependencies of the action, so there does not seem to be any fundamental obstacle to using them as dependencies.

Reply:
> Off-topic for the present PR, but with all the ongoing refactoring, is it/will it be possible to use these kinds of variables in the deps field? If memory serves they can already be used in action fields and they will be included automatically in the set of dependencies of the action, so there does not seem to be any fundamental obstacle to using them as dependencies.

That's indeed one of the goals. But it's a bit trickier than I thought. Consider a rule like this:

```
(rule
 (deps (:foo (glob_files %{read:dir}/*.ml)))
 (action (run ./bla.exe %{foo})))
```

To compile this rule, we need to build `dir`, evaluate the glob and then use it to evaluate the action. For that we need a primitive like:

```
val dyn_glob : File_selector.t Build.t -> Path.Set.t Build.t
```

I'm experimenting with adding something like that.